packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -212,7 +212,7 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     }
 
     public async isRoot(): Promise<boolean> {
-        return (await this.getInitialSnapshotDetails()).isRootDataStore;
+        return this.isRootDataStore || (await this.getInitialSnapshotDetails()).isRootDataStore;
```

Comment:
There should be a single source of truth for rootiness - `this.isRootDataStore`.
So, we should call `await thjis.getInitialSnapshotDetails()` and then return `this.isRootDataStore`. Basically, make sure that the initial snapshot details have been loaded before we return whether this is a root data store or not.

Reply:
Also, after this, `isRootDataStore` can now be removed from `ISnapshotDetails`.

Reply:
I would also add a comment here saying that `getInitialSnapshotDetails` has to be called before returning whether this is a root data store because the initial value is stored in the initial snapshot of the data store.